### PR TITLE
Bugfix/rerendering by hooks

### PR DESCRIPTION
**Pull request by anton-karlovskiy**
For this issue https://github.com/GoogleChromeLabs/adaptive-loading/issues/55
I updated the relevant hooks & their unit tests, demos and deployments.

### CODE REVIEW

**Comment by anton-karlovskiy** [comment]
@addyosmani
It's ready for review.

File: cra-memory-considerate-loading/src/utils/hooks.js [hunk 1]
```diff
@@ -14,25 +14,25 @@
...
-const unsupportMessage = 'The Memory Status API is not supported on this platform.';
+export const CLIENT_SIDE_UNSUPPORT_MESSAGE = 'The Memory Status API is not supported on this platform.';
+const SERVER_SIDE_UNSUPPORT_MESSAGE = 'This Memory React Hook uses Window interface so it\'s not possible to use this on Server Side Rendering';
 
 // Tune these for your application
-const MAX_MEMORY_LIMIT = 50 * 1048576; // 50MB
-const MAX_PERCENT_THRESHOLD = 90;
+export const MAX_MEMORY_LIMIT = 50 * 1048576; // 50MB
+export const MAX_PERCENT_THRESHOLD = 90;
+
```
Comment (addyosmani) [inline cra-memory-considerate-loading/src/utils/hooks.js]: I think we should avoid inline exporting values here (`export const <foo>`) and instead export these all as part of the final `export { useMemoryStatus };` line at the end. It gives us a central place to see what's actually being exported.

File: hooks/hardware-concurrency/index.js [hunk 2]
```diff
@@ -14,22 +14,21 @@
  * limitations under the License.
  */
 
-import { useState, useEffect } from 'react';
+import { useState } from 'react';
 
-export const unsupportMessage = 'The Hardware Concurrency API is not supported on this platform.';
+export const UNSUPPORT_MESSAGE = 'The Hardware Concurrency API is not supported on this platform.';
```
Comment (addyosmani) [inline hooks/hardware-concurrency/index.js]: Similar to my comment on the other file, can we centralize the exports at the end of the file along with `export { useHardwareConcurrency };`?

**Comment by anton-karlovskiy** [comment]
> Overall LGTM with minor nits

Let me check and update them shortly.

File: cra-memory-considerate-loading/src/utils/hooks.js (hunk 1 above)
Comment (anton-karlovskiy) [reply cra-memory-considerate-loading/src/utils/hooks.js]: Yes for sure. Sounds like a good idea.
Let me check other hooks as well and update them accordingly.

File: hooks/hardware-concurrency/index.js (hunk 2 above)
Comment (anton-karlovskiy) [reply hooks/hardware-concurrency/index.js]: Yes for sure.

**Comment by anton-karlovskiy** [comment]
@addyosmani 
FYI: I updated `export` positions as instructed.

**Comment by addyosmani** [comment]
Thanks, Anton!

**Pull request merged**